### ISSUE TITLE
Question: Source of Truth for Tests

### ISSUE THREAD

**Issue by jonatversprite** [nccgroup/Scout2#189]
What source(s) of truth are you using for this tool?  AWS Best Practices? Cloud Security Alliance?  CIS Benchmarks?  Experience?

**Comment by l01cd3v** [comment]
It started from a mix of AWS best practices and my own research, but now it's a mix of many things.

Whenever I need to look into a new service, I do my own research (go through best practices, benchmarks, ...) and also experience with as many settings as I can myself. I find that going through the process of creating resources is a good way to identify security-related settings. So far, I think this has been the best way for me to come up with new test cases.

I'm not a member of the Cloud Security Alliance so I've never looked at their resources. I've looked at the CIS benchmark as well and it's an overall pretty good source of truth (except for the monitoring section as this may be implemented in many ways without cloudwatch). I tried implementing as many checks as I could (from v1) in https://github.com/nccgroup/Scout2/blob/master/AWSScout2/rules/data/rulesets/cis-02-29-2016.json and will probably try to go through the latest version to create additional checks.

For the default ruleset, I try to find the right compromise between breadth of coverage and a limited number of false positives. There are a number of optional rules that may be enabled for specific use cases or tight controls (for example integration with managed IP ranges, documented there : https://blog.l01cd3v.fr/aws/2015/11/17/ip_ranges_management.html).
